src/atomate2/vasp/sets/core.py

```diff
@@ -775,6 +777,86 @@ def get_kpoints_updates(
         return {"reciprocal_density": self.reciprocal_density}
 
 
+@dataclass
+class MDSetGenerator(VaspInputSetGenerator):
+    """Class to generate VASP molecular dynamics input sets."""
+
+    def get_incar_updates(
+        self,
+        structure: Structure,
+        prev_incar: dict = None,
+        bandgap: float = 0,
+        vasprun: Vasprun = None,
+        outcar: Outcar = None,
+    ) -> dict:
+        """
+        Get updates to the INCAR for a molecular dynamics job.
+
+        Parameters
+        ----------
+        structure
+            A structure.
+        prev_incar
+            An incar from a previous calculation.
+        bandgap
+            The band gap.
+        vasprun
+            A vasprun from a previous calculation.
+        outcar
+            An outcar from a previous calculation.
+
+        Returns
+        -------
+        dict
+            A dictionary of updates to apply.
+        """
+        # TODO is is copied and changed based on pymatgen.io.vasp.sets.MPMDSet,
+        #  dobule check and discuss with others
+        updates = {
+            # expect to be changed by user
+            "NSW": 1000,
+            "ISIF": 0,
+            "MDALGO": 0,
+            "SMASS": 0,
+            "TEBEG": 300,
+            "TEEND": 300,
+            "POTIM": 2,
+            # exepct to be fixed
+            "PREC": "Normal",
+            "IBRION": 0,
+            "LREAL": "Auto",
+            "NELM": 500,
+            "NELMIN": 4,
+            "ISYM": 0,
+            "NBLOCK": 1,
+            "KBLOCK": 100,
+            "LSCALU": False,
+            "LCHARG": False,
+            "LPLANE": False,
+            "LWAVE": True,
+            "LDAU": False,
+            # "ISMEAR": 0,
+            # "EDIFF_PER_ATOM": 0.00001, # TODO deprecated?
+            # "MAXMIX": 20,
+            # "BMIX": 1,
+            # "NSIM": 4,
+            # "ADDGRID": True,
+        }
+
+        # use VASP default ENCUT
+        update["ENCUT"] = None
+
+        if defaults["ISPIN"] == 1:
```

Comment:
Not sure what defaults is here?

Instead of this, you might want to set `auto_ispin: bool = True` in this dataclass. See the docstring for that setting here: https://github.com/materialsproject/atomate2/blob/6445c2b67289abf9adc5a314aed0ae3488373a1d/src/atomate2/vasp/sets/base.py#L257

And this is an example of where I've turned it on: https://github.com/materialsproject/atomate2/blob/6445c2b67289abf9adc5a314aed0ae3488373a1d/src/atomate2/vasp/sets/core.py#L528